src/js/yeldaWebchatInjector.js

```diff
@@ -775,40 +775,109 @@ class YeldaChat {
+      return 'local'
+    }
+    return 'prod'
+  }
+
+  /**
+   * Get the chat bubble url base based on isFallback
+   * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {Boolean} isFallback fallback means yelda endpoint
+   * @returns {String} url base
+   */
+  getChatBubbleUrlBase(data, isFallback) {
+    if (isFallback) {
+      return `${data.assistantUrl}/assistants`
+    }
+    return 'https://webchat.yelda.ai/webchat'
+  }
+
   /**
    * return getAssistantSettings endpoint URL from data
    * @param {Object} data { assistantUrl, assistantId, settingId }
+   * @param {String} urlBase
    * @return {String} url
    */
-  getAssistantSettingsUrl(data) {
+  getAssistantSettingsUrl(data, urlBase) {
+    // The app env is required as query for the external chatbubble endpoint
+    const appEnv = this.getAppEnv(window.location.href)
+    const envQuery = `env=${appEnv}`
     if(data.settingId) {
-      return  `${data.assistantUrl}/assistants/settings/${data.settingId}/chatBubble`
+      return  `${urlBase}/settings/${data.settingId}/chatBubble?${envQuery}`
     }
 
-    return  `${data.assistantUrl}/assistants/${data.assistantId}/chatBubble/${data.locale}`
+    return  `${urlBase}/${data.assistantId}/chatBubble/${data.locale}?${envQuery}`
   }
 
   /**
    * Update assistantImage with assistant settings from backend if any
    * @param {Object} data { assistantUrl, assistantId, settingId }
    * @return {Promise}
    */
-  getAssistantSettings(data) {
+  getAssistantSettings(data, { isFallback = false, customTimeout = null } = {}) {
+    // Set `this` to be able to do some recursion in onreadystatechange
+    const self = this;
     return new Promise((resolve, reject) => {
       try {
         const xhr = new XMLHttpRequest()
-        const url = this.getAssistantSettingsUrl(data)
 
+        // Get the url base depending on if we are doing the basic request (external chatbubble endpoint) 
+        // or the fallback one (chatbubble endpoint in Yelda)
+        const urlBase = this.getChatBubbleUrlBase(data, isFallback)
+        const url = this.getAssistantSettingsUrl(data, urlBase)
+        
         xhr.open('GET', url)
+        if (customTimeout) {
+          xhr.timeout = customTimeout
+        }
         xhr.send()
 
         // Bind and call are necessary to pass the "this" to the callback function
         xhr.onreadystatechange = function() {
           if (xhr.readyState === 4) {
-            const webchatSettings = xhr.responseText ? JSON.parse(xhr.responseText).data : null
+            const responseText = xhr.responseText ? JSON.parse(xhr.responseText) : null
+            // @TODO check if we have still have case where the settings are set in a data field 
+            // like { data: { ...webchatSettings} } instead of { ...webchatSettings }
+            // From last tests, data is received as { ...webchatSettings } from both the external endpoint and the yelda endpoint
```

Comment:
https://app.yelda.ai/assistants/5cdaa648d721360f1b6a7109/chatBubble/fr_FR
there is a .data
![image](https://user-images.githubusercontent.com/5340081/150323761-82590e16-379e-4d0c-9a62-d6d539bd6888.png)

Why ? Because you tested with staging code, which is indeed broken !
checkout master branch and see : we return a not boomyfied endpoint with data !
```js
  async getFormattedAssistantChatBubbleSettings({ assistantId, settingId }, locale) {
    const assistantPlatformSettings = await this.getAssistantChatBubbleSettingsQuery({ assistantId, settingId }, locale)

    return {
      code: 20000,
      data: this.formatChatBubbleSettings(assistantPlatformSettings)
    }
  }
```

 SO, considering that clients do not update their chatplugin regularly for some, we should handle both format. 
 
 1. the new webchat endpoint can return with no data, because older chat plugin won't call it 
 2. the new webchat endpoint should still return the .data to not bread existing prod add 
 
 what do you think ?
 
 note that for the fallback case `const webchatSettings = responseText && (responseText.data || responseText)` is a good migration solution : once no more client with the old chat plugin version, we could remove it !
 
 
 :bulb: OR we can version the call : new endpoint call v1, and v1 return with no data
 :bulb: i would suggest to integrate in this call an optional chatpluginVersion string => thus, we could more easily see with client use which chatplugin version in prod ! => open an issue on yelda ?

Reply:
> i would suggest to integrate in this call an optional chatpluginVersion string => thus, we could more easily see with client use which chatplugin version in prod ! => open an issue on yelda ?

I don't think I understand what to do. 
* Where should I get the `chatpluginVersion`, in Yelda or directly in the chat plugin?
* By `this call`, I guess you mean the chat bubble call made in the chat-plugin?
* If we add it now, it would work only for the newer version of the chat-plugin, right?

Reply:
> Where should I get the chatpluginVersion, in Yelda or directly in the chat plugin?

to be done later, i opened https://github.com/Yeldaai/chat-plugin/issues/121
" directly in the chat plugin" => we would like to know which version are used

> By this call, I guess you mean the chat bubble call made in the chat-plugin?

yes

> If we add it now, it would work only for the newer version of the chat-plugin, right?

yes